sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+                                 ' smaller than the max'
+                                 ' n_features_to_select value.')
+
+        if isinstance(self.n_features_to_select, tuple):
+            select_in_range = True
+        else:
+            select_in_range = False
+            k_to_select = self.n_features_to_select
+
+        self._scorer = check_scoring(self.estimator, scoring=self.scoring)
+
+        cloned_estimator = clone(self.estimator)
+
+        self._n_features = X.shape[1]
+        self.subsets_ = {}
+        orig_set = set(range(self._n_features))
+        if self.forward:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[1]
+            k_idx = ()
+            k = 0
+        else:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[0]
+            k_idx = tuple(range(self._n_features))
+            k = len(k_idx)
+            k_score = self._calc_score(X, y, k_idx, cloned_estimator)
+            self.subsets_[k] = {
+                'feature_subset_idx': k_idx,
+                'cv_scores': k_score,
+                'avg_score': np.mean(k_score)
+                }
+
+        best_subset = None
+        k_score = 0
+
+        while k != k_to_select:
+            prev_subset = set(k_idx)
+            if self.forward:
+                k_idx, k_score, cv_scores = self._inclusion(
+                    orig_set=orig_set,
+                    subset=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+            else:
+                k_idx, k_score, cv_scores = self._exclusion(
+                    feature_set=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+
+            k = len(k_idx)
+            if k not in self.subsets_ or (self.subsets_[k]['avg_score'] <
+                                          k_score):
+                self.subsets_[k] = {
+                    'feature_subset_idx': k_idx,
+                    'cv_scores': cv_scores,
+                    'avg_score': k_score
+                }
+
+        if select_in_range:
+            max_score = float('-inf')
+            for k in self.subsets_:
+                if (k < self.n_features_to_select[0] or
+                        k > self.n_features_to_select[1]):
+                    continue
+                if self.subsets_[k]['avg_score'] > max_score:
+                    max_score = self.subsets_[k]['avg_score']
+                    best_subset = k
+            k_score = max_score
+            k_idx = self.subsets_[best_subset]['feature_subset_idx']
+
+        self.support_ = k_idx
+        self.support_ = self._get_support_mask()
+        self.score_ = k_score
+        return self
+
+    def _calc_score(self, X, y, indices, estimator):
+        if self.cv:
+            scores = cross_val_score(estimator,
```

Comment:
what sucks is that we can't pass in the Parallel object so it'll spin up all the forks in each iteration. Yuck. @lesteve, help!

Reply:
You mean that the parallelism happens inside cross_val_score instead of parallelizing the iterations over the feature subsets? Wouldn't be hard to change and was done in the original implementation in mlxtend meanwhile: https://github.com/rasbt/mlxtend/blob/master/mlxtend/feature_selection/sequential_feature_selector.py#L407 

Let me know if you prefer this version (reg. parallelism) in scikit-learn as well

Reply:
I'm not sure. I expect this could make things much faster in some setups (benchmark it? but on Windows too...). I think it would not hurt to duplicate some of the `cross_val_score` logic here., reusing `_fit_and_score`. I think the duplicated code would be quite minimal.

Reply:
As far as I remember, it sped up the mlxtend implementation quite a bit. But it really depends on so many variables, thus it's hard to really benchmark it for the average use case. I think the subset-parallelism version is almost always better because you have to spawn subprocesses less often. Also, if cv < n_processors, it would allow for utilizing more resources (except there are maybe just a handful of features, and in this case, performance wouldn't matter much). I will have a look at `_fit_and_score` when I rewrite this